runelite-client/src/main/java/net/runelite/client/RuneLite.java

```diff
@@ -212,6 +220,32 @@ public ClientUpdateCheckMode convert(String v)
 			}
 		});
 
+		SwingUtilities.invokeAndWait(() ->
+		{
+			RuneLiteSplashScreen.setTheme();
+			if (options.has("no-splash"))
+			{
+				SPLASH_SCREEN = new RuneLiteSplashScreen(new File(LOGS_DIR, "client.log"));
```

Comment:
this new File(LOGS_DIR, "client.log") should be just LOG_FILE constant at top

Reply:
also, honestly, these constructors do not make sense.Why is file not at first place in both scenarios, and why no-splash still needs splash screen, and why adding string to splash screen constructor changes behaviour of it?

Reply:
I create the splash screen even when no-splash is specified so I didn't have to null-check SPLASH_SCREEN at every `SPLASH_SCREEN.setMessage()`. It also lets me use the splash screens dialog popups which i think should be displayable regardless of if the splash screen is being displayed.

Reply:
I would rather do proxy method and null checks. If you want just the dialog popups, separate them from splash screen